x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportUpdateFilterAction.java

```diff
@@ -0,0 +1,181 @@
+        indexUpdatedFilter(updatedFilter, filterWithVersion.version, request, listener);
+    }
+
+    private void indexUpdatedFilter(MlFilter filter, long version, UpdateFilterAction.Request request,
+                                    ActionListener<PutFilterAction.Response> listener) {
+        IndexRequest indexRequest = new IndexRequest(MlMetaIndex.INDEX_NAME, MlMetaIndex.TYPE, filter.documentId());
+        indexRequest.version(version);
+        indexRequest.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+
+        try (XContentBuilder builder = XContentFactory.jsonBuilder()) {
+            ToXContent.MapParams params = new ToXContent.MapParams(Collections.singletonMap(MlMetaIndex.INCLUDE_TYPE_KEY, "true"));
+            indexRequest.source(filter.toXContent(builder, params));
+        } catch (IOException e) {
+            throw new IllegalStateException("Failed to serialise filter with id [" + filter.getId() + "]", e);
+        }
+
+        executeAsyncWithOrigin(client, ML_ORIGIN, IndexAction.INSTANCE, indexRequest, new ActionListener<IndexResponse>() {
+            @Override
+            public void onResponse(IndexResponse indexResponse) {
+                jobManager.notifyFilterChanged(filter, request.getAddItems(), request.getRemoveItems());
+                listener.onResponse(new PutFilterAction.Response(filter));
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                Exception reportedException;
+                if (e instanceof VersionConflictEngineException) {
+                    reportedException = ExceptionsHelper.conflictStatusException("Error updating filter with id [" + filter.getId()
+                            + "] because it was modified while the update was in progress", e);
+                } else {
+                    reportedException = ExceptionsHelper.serverError("Error updating filter with id [" + filter.getId() + "]", e);
+                }
+                listener.onFailure(reportedException);
+            }
+        });
+    }
+
+    private void getFilterWithVersion(String filterId, ActionListener<FilterWithVersion> listener) {
+        GetRequest getRequest = new GetRequest(MlMetaIndex.INDEX_NAME, MlMetaIndex.TYPE, MlFilter.documentId(filterId));
+        executeAsyncWithOrigin(client, ML_ORIGIN, GetAction.INSTANCE, getRequest, new ActionListener<GetResponse>() {
+            @Override
+            public void onResponse(GetResponse getDocResponse) {
+                try {
+                    if (getDocResponse.isExists()) {
+                        BytesReference docSource = getDocResponse.getSourceAsBytesRef();
+                        try (InputStream stream = docSource.streamInput();
+                             XContentParser parser =
+                                     XContentFactory.xContent(getDocResponse.getSourceAsBytes())
```

Comment:
Is this method not deprecated? I think it should be 
`XContentParser parser = XContentFactory.xContent(XContentType.JSON)
                                     .createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE, stream))`